src/main/java/by/dragonsurvivalteam/dragonsurvival/client/handlers/ClientFlightHandler.java

```diff
@@ -247,241 +252,233 @@ private static void spawnSpinParticle(Player player, ParticleOptions particleDat
 
-								Vec3 moveVector = ClientDragonRender.getInputVector(new Vec3(movement.leftImpulse, 0, movement.forwardImpulse), 1F, player.yRot);
-								moveVector.multiply(1.3, 0, 1.3);
-
-								double lookY = lookVec.y;
-
-								boolean moving = movement.up || movement.down || movement.left || movement.right;
-								double yaw = Math.toRadians(player.yHeadRot + 90);
+									if (deltaMovement.y < 0 && horizontalView > 0) {
+										double downwardMomentum = deltaMovement.y * -0.1D * (double) verticalDelta;
+										deltaMovement = deltaMovement.add(viewVector.x * downwardMomentum / horizontalView, downwardMomentum, viewVector.z * downwardMomentum / horizontalView);
+									}
 
-								if(ServerFlightHandler.isSpin(player)){
-									ax += Math.cos(yaw) / 500 * 200;
-									az += Math.sin(yaw) / 500 * 200;
-									ay = lookVec.y / 8;
-								}
+									if (pitch < 0 && horizontalView > 0) {
+										// Handle movement when the player makes turns
+										double delta = horizontalMovement * -Mth.sin(pitch) * 0.04D;
+										deltaMovement = deltaMovement.add(-viewVector.x * delta / horizontalView, delta * 3.2D, -viewVector.z * delta / horizontalView);
+									}
 
-								if(ServerFlightHandler.stableHover && !movement.jumping && !movement.shiftKeyDown && !ServerFlightHandler.isSpin(player) && !ServerFlightHandler.isGliding(player)){
-									ay = Math.max(ay, g * 1.1);
-								}
+									if (horizontalView > 0) {
+										deltaMovement = deltaMovement.add((viewVector.x / horizontalView * horizontalMovement - deltaMovement.x) * 0.1D, 0.0D, (viewVector.z / horizontalView * horizontalMovement - deltaMovement.z) * 0.1D);
+									}
 
-								if(moving && !movement.jumping && !movement.shiftKeyDown){
-									maxForward = 0.8;
-									moveVector.multiply(1.4, 0, 1.4);
-									motion = new Vec3(Mth.lerp(0.1, motion.x, moveVector.x), 0, Mth.lerp(0.1, motion.z, moveVector.z));
-									motion = new Vec3(Mth.clamp(motion.x, -maxForward, maxForward), 0, Mth.clamp(motion.z, -maxForward, maxForward));
+									// Increase speed while flying down or height when flying up
+									if (viewVector.y < 0) {
+										ax += Math.cos(yaw) / 500;
+										az += Math.sin(yaw) / 500;
+									} else {
+										ax *= 0.99;
+										az *= 0.99;
+										ay = viewVector.y / 8; // TODO :: Causes flying dragons to gain height when they look up but not moving otherwise?
+									}
 
-									motion = motion.add(ax, ay, az);
+									double speedLimit = ServerFlightHandler.maxFlightSpeed;
+									ax = Mth.clamp(ax, -0.2 * speedLimit, 0.2 * speedLimit);
+									az = Mth.clamp(az, -0.2 * speedLimit, 0.2 * speedLimit);
 
-									ax *= 0.9F;
-									ay *= 0.9F;
-									az *= 0.9F;
+									if (ServerFlightHandler.isSpin(player)) { // TODO :: If the spin move is used in water won't the acceleration be applied twice?
```

Comment:
The acceleration in the water is noticeably less noticeable, but I like the current velocity